plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -233,15 +235,26 @@
                 domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
                 "This PredicatePushdownController can be used only for chars and varchars");
 
-        if (domain.isOnlyNull() ||
-                // PostgreSQL is case sensitive by default
-                domain.getValues().isDiscreteSet()) {
+        if (domain.isOnlyNull()) {
             return FULL_PUSHDOWN.apply(session, domain);
         }
 
+        // PostgreSQL is case sensitive by default
         // PostgreSQL by default orders lowercase letters before uppercase, which is different from Trino
         // TODO We could still push the predicates down if we could inject a PostgreSQL-specific syntax for selecting a collation for given comparison.
-        return DISABLE_PUSHDOWN.apply(session, domain);
+        if (!domain.getValues().isDiscreteSet()) {
```

Comment:
Is there any way to disable pushdown for discrete sets but only when an inequality is involved? This condition is as good as turning off pushdown completely.

Reply:
I'm a bit confused now. Discrete set means independent values compared using either equality or inequality? Correct?

If so, then why will there be incorrect results? PostgreSQL is case-sensitive and so is Trino. Only the sort order is different so this should only affect inequality or ranges? Equality conditions on discrete sets should give correct results?

Reply:
I think I understand now. A comment like ``predicate pushdown could return incorrect results for operators like `<` or `>` `` would help.

Reply:
I've tweaked the comments a bit, hopefully it should be clearer now.
`!domain.getValues().isDiscreteSet()` should catch only range predicates